include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -547,4 +547,36 @@ def TTNN_DeallocOp : TTNN_Op<"dealloc"> {
     let arguments = (ins AnyRankedTensor:$input);
 }
 
+def TTNN_AllGatherOp: TTNN_Op<"all_gather"> {
+    let summary = "All gather op.";
+    let description = [{
+        Tensor All Gather operation
+    }];
+
+    let arguments = (ins AnyRankedTensor:$input,
+                         I32Attr:$dim,
+                         I32Attr:$num_links);
+//                         OptionalAttr<TTNN_MemoryConfigAttr>:$memory_config);
```

Comment:
Should we keep this under comment?

Reply:
It will be an optional attribute. I intended to add it later.

Reply:
I removed the commented code. Will add the optional attribute `memory_config` later.